crates/env/src/call/call_builder.rs

```diff
@@ -341,7 +343,7 @@ where
 /// The default call type for cross-contract calls. Performs a cross-contract call to `callee`
 /// with gas limit `gas_limit`, transferring `transferred_value` of currency.
 pub struct Call<E: Environment> {
-    callee: E::AccountId,
+    callee: Option<E::AccountId>,
```

Comment:
I think this should remain `E::AccountId` so a `Call` can only ever be constructed with a callee. So we could pass it to the constructor e.g. `Call::new(callee)` instead of `Call::new().callee(callee)`. Or even modify the `CallBuilder` so that we don't need to construct `Call` directly.

We should prevent the user constructing invalid states using the type system where possible.

Reply:
e.g. https://github.com/paritytech/ink/pull/1636